units/generate_tasks.go

```diff
@@ -66,7 +68,15 @@ func NewGenerateTasksJob(versionID, taskID string, ts string) amboy.Job {
 }
 
 func (j *generateTasksJob) generate(ctx context.Context, t *task.Task) error {
-	start := time.Now()
+	ctx, span := tracer.Start(ctx, "task-generation", trace.WithAttributes(
+		attribute.String(evergreen.TaskIDOtelAttribute, t.Id),
+		attribute.Int(evergreen.TaskExecutionOtelAttribute, t.Execution),
+		attribute.String(evergreen.VersionIDOtelAttribute, t.Version),
+		attribute.String(evergreen.BuildIDOtelAttribute, t.BuildId),
+		attribute.String(evergreen.ProjectIDOtelAttribute, t.Project),
+		attribute.String(evergreen.VersionRequesterOtelAttribute, t.Requester),
+	))
+	defer span.End()
```

Comment:
Idk if it's desirable, but if we use [ContextWithAttributes](https://github.com/evergreen-ci/utility/blob/4d91d64f59fb653893f0f0b0f301971ce050b389/otel.go#L37-L41) it could make all the child spans have all these attributes. For example, you could query how long `add-new-builds` takes by project.

Reply:
Oh that's great, thanks!